src/main/java/net/glowstone/command/minecraft/FillCommand.java

```diff
@@ -0,0 +1,249 @@
+            CommandMessages commandMessages) {
+        if (!testPermission(sender, commandMessages.getPermissionMessage())) {
+            return true;
+        }
+        if (args.length < 7) {
+            sendUsageMessage(sender, commandMessages);
+            return false;
+        }
+        String itemName = CommandUtils.toNamespaced(args[6].toLowerCase());
+        Material type = ItemIds.getBlock(itemName);
+        if (type == null) {
+            new LocalizedStringImpl("setblock.invalid.type",
+                    commandMessages.getResourceBundle())
+                    .sendInColor(ChatColor.RED, sender, itemName);
+            return false;
+        }
+        byte dataValue = 0;
+        if (args.length > 7) {
+            String state = args[7];
+            BlockStateData data = CommandUtils.readState(sender, type, state);
+            if (data == null) {
+                return false;
+            }
+            if (data.isNumeric()) {
+                dataValue = data.getNumericValue();
+            } else {
+                try {
+                    dataValue = StateSerialization.parseData(type, data).getData();
+                } catch (InvalidBlockStateException e) {
+                    sender.sendMessage(ChatColor.RED + e.getMessage());
+                    return false;
+                }
+            }
+        }
+        Location location1 = CommandUtils.getLocation(sender, args[0], args[1], args[2]);
+        Location location2 = CommandUtils.getLocation(sender, args[3], args[4], args[5]);
+        double x1 = location1.getX();
+        double y1 = location1.getY();
+        double z1 = location1.getZ();
+        double x2 = location2.getX();
+        double y2 = location2.getY();
+        double z2 = location2.getZ();
+        double xmin = 0;
+        double ymin = 0;
+        double zmin = 0;
+        double xmax = 0;
+        double ymax = 0;
+        double zmax = 0;
+        double placeX;
+        double placeY;
+        double placeZ;
+        if (x1 > x2) {
+            xmin = x2;
+            xmax = x1;
+        } else {
+            xmax = x2;
+            xmin = x1;
+        }
+        if (y1 > y2) {
+            ymin = y2;
+            ymax = y1;
+        } else {
+            ymax = y2;
+            ymin = y1;
+        }
+        if (z1 > z2) {
+            zmin = z2;
+            zmax = z1;
+        } else {
+            zmax = z2;
+            zmin = z1;
+        }
+        double blockAmount = (xmax - xmin) * (ymax - ymin) * (zmax - zmin);
+        if (blockAmount > 40000) {
+            new LocalizedStringImpl("fill.tooManyBlocks",
+            commandMessages.getResourceBundle())
+                .sendInColor(ChatColor.RED, sender, Double.toString(blockAmount));
+            return false;
+        }
+        int placementMode = 0;
+        if (args.length > 8) {
+            if (args[8].equalsIgnoreCase("outline")) {
+                placementMode = 1;
```

Comment:
Replace these magic values with an enum, then have a lookup table to convert the command argument to enum values.